Code/Light.GuardClauses.Tests/ComparableAssertions/RangeTests.cs

```diff
@@ -49,4 +54,116 @@ public static void ConstructorException(int from, int to)
         act.Should().Throw<ArgumentOutOfRangeException>().And
            .Message.Should().Contain($"{nameof(to)} must not be less than {from}, but it actually is {to}.");
     }
+
+    [Theory]
+    [MemberData(nameof(Collections))]
+    public static void RangeForCollections(IEnumerable enumerable)
+    {
+        // ReSharper disable PossibleMultipleEnumeration
+        var range = Range.For(enumerable);
+
+        var expectedRange = new Range<int>(0, enumerable.Count(), true, false);
+        range.Should().Be(expectedRange);
+        // ReSharper restore PossibleMultipleEnumeration
+    }
+
+    public static readonly TheoryData<IEnumerable> Collections =
+        new ()
+        {
+            new List<int> { 1, 2, 3, 4 },
+            "This is a long string",
+            new[] { 'a', 'b', 'c', 'd' },
+            new ObservableCollection<long> { 1, -1 },
+            new ArrayList()
+        };
+
+    [Fact]
+    public static void EnumerableNull()
+    {
+        // ReSharper disable once RedundantCast -- I want to specifically target the overload accepting IEnumerable
+        var act = () => Range.For(((IEnumerable) null)!);
+
+        act.Should().Throw<ArgumentNullException>()
+           .And.ParamName.Should().Be("enumerable");
```

Comment:
Hint: FluentAssertions has a shortcut for checking the parameter name of an `ArgumentException`: `.WithParameterName("enumerable")`

Reply:
Fixed in ca5d82d4cc56348defd7a8f8edf3380af1ea0b2d